src/odemis/driver/xt_client.py

```diff
@@ -800,31 +800,42 @@ def get_mpp_orientation(self):
             self.server._pyroClaimOwnership()
             return self.server.get_mpp_orientation()
 
+    def mpp_orientation_info(self):
+        """
+        Get the current multi probe orientation information, contains the range and the unit (defined as degrees).
+
+        :return (dict str -> list (of length 2)): The range of the rotation for the keyword "range"
+        and the 'degrees' for the keyword "unit"
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.mpp_orientation_info()
+
     def get_aperture_index(self):
         """
         Get the current aperture index.
 
-        :return (float): Aperture index (range 0 - 14)
+        :return (int): Aperture index (typical range 0 - 14)
         """
         with self._proxy_access:
             self.server._pyroClaimOwnership()
-            return self.server.get_aperture_index()
+            return int(self.server.get_aperture_index())
 
     def set_aperture_index(self, aperture_idx):
         """
         Set the current aperture index.
 
-        :param aperture_idx  (float): Aperture index (range 0 - 14)
+        :param aperture_idx (float): Aperture index (typical range 0 - 14)
```

Comment:
Should the input here not be an int then as well as for the return value of the corresponding get method? It seems to be more logic to me that the return value of the get method and the input value of the set method are of the same type.

Reply:
Correct, fixed!